### PR TITLE
Added tests for non-event consumes ordering of modules

### PR DESCRIPTION
**Pull request by Dr15Jones**
#### PR description:

Added tests and a test module to validate the ordering of modules called based on data dependencies.
#### PR validation:

The new unit tests run and pass.

### CODE REVIEW

**Comment by Dr15Jones** [comment]
please test

**Comment by Dr15Jones** [comment]
@makortel @wddgit please review

File: FWCore/Framework/test/stubs/ToyIntProducers.cc [hunk 1]
```diff
@@ -486,6 +486,100 @@ namespace edmtest {
...
+    edm::EDGetTokenT<IntProduct> brGet_;
+    edm::EDGetTokenT<IntProduct> blGet_;
+    edm::EDGetTokenT<IntProduct> elGet_;
+    edm::EDGetTokenT<IntProduct> erGet_;
+  };
+
+  void NonEventIntProducer::accumulate(edm::StreamID iID, edm::Event const& e, edm::EventSetup const&) const {}
+  void NonEventIntProducer::globalBeginRunProduce(edm::Run& r, edm::EventSetup const&) const {
+    if (not brGet_.isUninitialized()) {
+      r.get(brGet_);
```
Comment (makortel) [inline FWCore/Framework/test/stubs/ToyIntProducers.cc]: Would it be worth of the effort to check that the read value actually is what is expected?

File: FWCore/Framework/test/stubs/ToyIntProducers.cc [hunk 2]
```diff
@@ -486,6 +486,100 @@ namespace edmtest {
     }
   };
 
+  //--------------------------------------------------------------------
+
+  class NonEventIntProducer : public edm::global::EDProducer<edm::Accumulator,
```
Comment (makortel) [inline FWCore/Framework/test/stubs/ToyIntProducers.cc]: Would it be worth of the effort to test also stream, one, and limited?

Comment (Dr15Jones) [reply FWCore/Framework/test/stubs/ToyIntProducers.cc]: The internals of the framework treat all kinds of modules identically when it comes to launching work at non-event transitions.

File: FWCore/Framework/test/stubs/ToyIntProducers.cc (hunk 1 above)
Comment (Dr15Jones) [reply FWCore/Framework/test/stubs/ToyIntProducers.cc]: There are many other tests which check that one gets back the actual data one requested. We could do the same here. I'm OK either way.

**Comment by wddgit** [comment]
Looks good to me. Good test to add. Two minor comments, feel free to ignore.
1. You could have a default value for the ivalue parameter in fillDescriptions since you do not check the values and do not care what the value is. Then you do not need to bother setting the parameter in the cfg files.
2. I can see the value in keeping the 4 cfg's separate and maybe it is worth it, but one might combine the 4 cfg's into 1 cfg to make the unit tests run a little faster. On the other hand, each individual test is simpler the way it is...

**Comment by Dr15Jones** [comment]
please test

**Comment by Dr15Jones** [comment]
+1

**Comment by fabiocos** [comment]
+1

comments raised in the review have been addressed, for the time being I understand that separate cfgs are preferred

**Pull request merged**